### PR TITLE
Windows sys/meson.py: Add r2 shell script

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

During Windows installation, this pr adds an `r2` shell script that calls `radare2`, for shells that don't recognize .bat or .cmd extensions (e.g. git bash). This is to support #16747.

**Test plan**

Running `r2` on the git bash prompt works (assuming that it's in PATH ofc).

**Closing issues**

...

### CODE REVIEW

File: sys/meson.py [hunk 1]
```diff
@@ -172,6 +172,11 @@ def win_dist(args):
     with open(r2_bat_fname, 'w') as r2_bat:
         r2_bat.write('@"%~dp0\\radare2" %*\n')
 
+    r2_sh_fname = args.install + r'\bin\r2'
```
Comment (radare) [inline sys/meson.py:175]: uhm isnt this meson.py suposed to run on windows too?

File: sys/meson.py [hunk 2]
```diff
@@ -172,6 +172,11 @@ def win_dist(args):
...
+    log.debug('create "%s"', r2_sh_fname)
+    with open(r2_sh_fname, 'w') as r2_sh:
+        r2_sh.write('#!/bin/sh\n$(dirname "$0")/radare2 "$@"\n')
```
Comment (radare) [inline sys/meson.py:178]: why not just use a symlink?

File: sys/meson.py (hunk 1 above)
Comment (kazarmy) [reply sys/meson.py:175]: it does ... this pr is for people running POSIX shells on Windows (e.g. the bash shell that comes with [MSYS2](https://www.msys2.org/))

File: sys/meson.py (hunk 2 above)
Comment (kazarmy) [reply sys/meson.py:178]: on Windows, symlinks require their creators to jump through hoops ([either Administrator or Developer Mode](https://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/)) and i don't think `sys/meson.py` can assume that either is enabled

**Pull request merged**